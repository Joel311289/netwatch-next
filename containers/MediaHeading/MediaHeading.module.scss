.wrapper {
  border-left: 3px solid var(--color-primary);
  color: var(--color-text);
  padding-left: 10px;
  height: 28px;
  margin: 10px 0;
}

.text {
  font-size: 1.1rem;
  font-weight: 600;
}

.icon {
  margin-top: 2.5px;
  margin-left: 5px;
  font-size: 1.8rem;
}

.wrapper:hover .icon {
  color: var(--color-primary);
}
