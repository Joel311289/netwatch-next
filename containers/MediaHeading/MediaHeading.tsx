import { ElementChildren, ElementHTML, ElementLink, ElementSkeleton } from '@/types';
import { Space } from '@/components/layout';
import { FiChevronRight } from 'react-icons/fi';
import { withNavigation } from '@/hoc/withNavigation';
import { withSkeleton } from '@/hoc/withSkeleton';
import { Heading } from '@/components/typography';
import { classes } from '@/utils/helpers';
import styles from '@/containers/MediaHeading/MediaHeading.module.scss';

export type MediaHeadingProps = typeof defaultProps &
  ElementHTML &
  ElementChildren<string> &
  ElementSkeleton &
  ElementLink;

const defaultProps = {};

const MediaHeading = ({ className, children, skeleton, href }: MediaHeadingProps) => {
  return (
    <Space justify="between" align="center" gap={20} className={classes(styles.wrapper, className)}>
      <Space align="center">
        <Heading skeleton={skeleton} level={3}>
          {children}
        </Heading>

        {href && <FiChevronRight className={styles.icon} />}
      </Space>
    </Space>
  );
};

MediaHeading.defaultProps = defaultProps;

const MediaHeadingWithNavigation = withNavigation<MediaHeadingProps>(MediaHeading);
const MediaHeadingWithSkeleton = withSkeleton<MediaHeadingProps>(MediaHeadingWithNavigation, {
  variant: 'heading'
});

export default MediaHeadingWithSkeleton;
