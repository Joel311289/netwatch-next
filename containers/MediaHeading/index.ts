export { default as MediaHeading } from '@/containers/MediaHeading/MediaHeading';
