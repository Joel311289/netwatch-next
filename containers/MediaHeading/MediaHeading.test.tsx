import { render, screen } from '@testing-library/react';
import MediaHeading from '@/containers/MediaHeading/MediaHeading';

describe('Tests MediaHeading component', () => {
  test('renders component correctly', () => {
    const { container } = render(<MediaHeading>Test</MediaHeading>);

    expect(container).toMatchSnapshot();
  });

  test('renders children correctly', () => {
    render(<MediaHeading>Test</MediaHeading>);

    expect(screen.getByText('Test')).toBeInTheDocument();
  });
});
