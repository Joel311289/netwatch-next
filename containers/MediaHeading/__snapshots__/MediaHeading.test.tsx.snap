// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests MediaHeading component renders component correctly 1`] = `
<div>
  <div
    class="wrapper wrapper"
  >
    <div
      class="wrapper"
    >
      <h3
        class="wrapper"
      >
        Test
      </h3>
      <style>
        
        .wrapper {
          display: flex;
          flex-direction: row;
          flex-wrap: nowrap;
          
          
          align-items: center;
        }
      
      </style>
    </div>
    <style>
      
        .wrapper {
          display: flex;
          flex-direction: row;
          flex-wrap: nowrap;
          gap: 20px;
          justify-content: space-between;
          align-items: center;
        }
      
    </style>
  </div>
</div>
`;
